File: samples/client/3_1_0_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/CustomIsoparser.java
package org.openapijsonschematools.client.schemas.validation;

import java.time.ZonedDateTime;
import java.time.LocalDate;

public final class CustomIsoparser {

    public ZonedDateTime parseIsodatetime(String dateTime) {
        return ZonedDateTime.parse(dateTime);
    }

    public LocalDate parseIsodate(String date) {
        return LocalDate.parse(date);
    }

}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/CustomIsoparser.kt
package org.openapijsonschematools.client.schemas.validation

import java.time.LocalDate
import java.time.ZonedDateTime

class CustomIsoparser {
    companion object {
        fun parseIsodatetime(dateTime: String): ZonedDateTime {
            return ZonedDateTime.parse(dateTime)
        }

        fun parseIsodate(date: String): LocalDate {
            return LocalDate.parse(date)
        }
    }
}